File: 110_Balanced_Binary_Tree/Solution.py
class Solution(object):
    def isBalanced(self, root):
        return self.obter_altura(root) != -1

    def obter_altura(self, no):
        if not no:
            return 0

        altura_esq = self.obter_altura(no.left)
        if altura_esq == -1:
            return -1

        altura_dir = self.obter_altura(no.right)
        if altura_dir == -1:
            return -1

        if abs(altura_esq - altura_dir) > 1:
            return -1

        return max(altura_esq, altura_dir) + 1

File: 110_Balanced_Binary_Tree/Solution.java
class Solution {
    public boolean isBalanced(TreeNode root) {
        return altura(root) != -1;
    }

    private int altura(TreeNode no) {
        if (no == null) return 0;

        int alturaEsquerda = altura(no.left);
        if (alturaEsquerda == -1) return -1; 

        int alturaDireita = altura(no.right);
        if (alturaDireita == -1) return -1; 

        if (Math.abs(alturaEsquerda - alturaDireita) > 1) return -1;

        return Math.max(alturaEsquerda, alturaDireita) + 1;
    }
}